caravel/assets/javascripts/SqlLab/components/TabbedSqlEditors.jsx

```diff
@@ -1,14 +1,60 @@
 import React from 'react';
-import { DropdownButton, MenuItem, Tab, Tabs } from 'react-bootstrap';
+import { DropdownButton, MenuItem, Tab, Tabs, Popover, OverlayTrigger } from 'react-bootstrap';
 import { connect } from 'react-redux';
 import { bindActionCreators } from 'redux';
+import CopyToClipboard from '../../components/CopyToClipboard';
 import * as Actions from '../actions';
 import SqlEditor from './SqlEditor';
 import shortid from 'shortid';
 
 let queryCount = 1;
 
 class QueryEditors extends React.Component {
+  componentWillMount() {
+    const uri = window.location.toString();
+    const cleanUri = uri.substring(0, uri.indexOf('?'));
+    const query = window.location.search.substring(1);
+
+    if (query) {
+      queryCount++;
+      const qe = {
```

Comment:
could rename to `queryEditorProps` for clarity. i forget what we mean by `qe` sometimes.